esp-hal-common/src/soc/mod.rs

```diff
@@ -11,6 +13,60 @@ mod soc;
 
 mod efuse_field;
 
+// Indicates the state of setting the mac address
+// 0 -- unset
+// 1 -- in the process of being set
+// 2 -- set
+//
+// Values other than 0 indicate that we cannot attempt setting the mac address
+// again, and values other than 2 indicate that we should read the mac address
+// from eFuse.
+static MAC_OVERRIDE_STATE: AtomicU8 = AtomicU8::new(0);
+static mut MAC_OVERRIDE: [u8; 6] = [0; 6];
+
+#[derive(PartialEq, Eq, Copy, Clone, Debug)]
+pub enum SetMacError {
+    AlreadySet,
+}
+
+impl soc::efuse::Efuse {
+    /// Set the base mac address
+    ///
+    /// The new value will be returned by `read_mac_address` instead of the one
+    /// hard-coded in eFuse. This does not persist across device resets.
+    ///
+    /// Can only be called once. Returns `Err(`[`SetMacError::AlreadySet`]`)`
+    /// otherwise.
+    pub fn set_mac_address(mac: [u8; 6]) -> Result<(), SetMacError> {
```

Comment:
Can we please avoid `get_`? It's generally [not recommended](https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter).

I would also like to shamelessly plug my [atomic_enum](https://crates.io/crates/portable_atomic_enum) fork that may look a bit nicer than magic numbers.